src/CloudNative.CloudEvents/Core/BinaryDataUtilities.cs

```diff
@@ -41,5 +44,59 @@ public static byte[] ToByteArray(Stream stream)
             stream.CopyTo(memory);
             return memory.ToArray();
         }
+
+        /// <summary>
+        /// Returns a read-only <see cref="MemoryStream"/> view over the given memory.
+        /// </summary>
+        /// <param name="memory">The memory to create a stream view over.</param>
+        /// <returns>A read-only stream view over <paramref name="memory"/>.</returns>
+        public static MemoryStream AsStream(ReadOnlyMemory<byte> memory)
+        {
+            var segment = GetArraySegment(memory);
+            return new MemoryStream(segment.Array, segment.Offset, segment.Count, false);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="memory"></param>
+        /// <param name="encoding"></param>
+        public static string GetString(ReadOnlyMemory<byte> memory, Encoding encoding)
+        {
+            // TODO: Use encoding.GetString(memory.Span) where we can?
```

Comment:
Good idea - looks like it is available in NetStandard 2.1.

Reply:
That means multi-targeting though... which adds more complexity when trying to diagnose things. Let's think about it.